packages/ethers/src/client.ts

```diff
@@ -319,6 +316,47 @@ export class Web3Modal extends Web3ModalScaffold {
           // eslint-disable-next-line no-negated-condition
         } else if (providerType === ConstantsUtil.AUTH_CONNECTOR_ID) {
           await this.authProvider?.disconnect()
+        } else if (providerType === ConstantsUtil.EIP6963_CONNECTOR_ID && provider) {
+          try {
+            const permissions = (await provider.request({ method: 'wallet_getPermissions' })) as {
```

Comment:
Don't we have the types of the response here? Why linter showing that assertion is unnecessary? If we don't know the return type, maybe we should override the lint for these lines? @tomiir @svenvoskamp

Reply:
Nope, the return type is a generic Promise<any> so ts can't infer the exact type of the response.
```js
request(req: W3mFrameTypes.RPCRequest): Promise<any>;
```

Also, i dont see the linting issue. its TS complaining in the line below `'permissions' is of type 'unknown'.`

Reply:
Nvm, I just ran the linter and saw the issue

Reply:
Fixed it, will wait for others to reply before resolving the conversation.